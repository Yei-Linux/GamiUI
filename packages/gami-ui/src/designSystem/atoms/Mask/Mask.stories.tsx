import React from 'react'
import { Meta } from '@storybook/react'

import Mask from '.'

export default {
  title: 'Atoms/Mask',
  component: Mask,
  args: {},
  argTypes: {},
} as Meta

export const Basic = (args: any) => <Mask {...args} />
Basic.args = {}
