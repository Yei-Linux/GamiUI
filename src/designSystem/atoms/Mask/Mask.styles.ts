import styled from '@emotion/styled'

export const MaskWrapper = styled.div<{ zIndex?: number; background?: string }>`
  background-color: ${(props) => props.background};
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  z-index: ${(props) => props.zIndex};
  height: 100%;
  position: fixed;
`
