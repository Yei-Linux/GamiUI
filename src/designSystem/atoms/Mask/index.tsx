export { default } from './Mask'
export { options } from './constants'
