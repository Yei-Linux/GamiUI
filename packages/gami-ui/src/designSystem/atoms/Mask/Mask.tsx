import React from 'react'
import { zIndex as zIndexToken } from 'styles/tokens'
import * as S from './Mask.styles'

export type TZIndex = 0 | 1 | 2 | 3 | 4 | 5

export interface IMask {
  /**
   * ZIndex of Layer Mask
   */
  zIndex?: TZIndex

  /**
   * Background Mask
   */
  background?: string

  /**
   * Action Click on Mask
   */
  onClick?: () => void
}

const Mask = ({
  zIndex = 1,
  onClick,
  background = 'rgba(0, 0, 0, 0.45)',
}: IMask) => {
  return (
    <S.Mask
      onClick={onClick}
      $background={background}
      $zIndex={zIndexToken[zIndex] as TZIndex}
    />
  )
}

export default Mask
