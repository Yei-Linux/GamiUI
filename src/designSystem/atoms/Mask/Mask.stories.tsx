import React from 'react'
import { Meta } from '@storybook/react'

import Mask from '.'

export default {
  title: 'Atoms/Mask',
  component: Mask,
  args: { ...Mask.defaultProps },
  argTypes: {},
} as Meta

export const Basic = (args: any) => <Mask {...args} />
Basic.args = {}
