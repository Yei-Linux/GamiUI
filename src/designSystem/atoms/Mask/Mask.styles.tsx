import styled from '@emotion/styled'
import { sizes } from 'styles/tokens'
import { TZIndex } from './Mask'

export const Mask= styled.div<{
  $zIndex: TZIndex
  $background: string
}>`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;

  height: ${sizes.height.full};

  background-color: ${({ $background }) => $background};
  z-index: ${({ $zIndex }) => $zIndex};
`
