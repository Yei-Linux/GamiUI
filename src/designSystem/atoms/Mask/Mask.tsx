import React from 'react'
import { MaskWrapper } from './Mask.styles'

export interface IMask {
  zIndex?: number
  onClick?: ()=> any
}

const Mask = ({ zIndex, onClick }: IMask) => {
  return <MaskWrapper onClick={onClick} zIndex={zIndex}/>
}

Mask.defaultProps = {
  zIndex: 1
}

export default Mask
