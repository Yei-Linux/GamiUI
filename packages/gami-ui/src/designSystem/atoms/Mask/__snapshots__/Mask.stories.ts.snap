// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Mask Basic 1`] = `
<div
  className="css-1528g5q-MaskWrapper e1qqpb1u0"
/>
`;
